semgrep/semgrep/autofix.py

```diff
@@ -20,28 +21,52 @@ def __init__(self, fixed_contents: str, fixed_lines: List[str]):
         self.fixed_lines = fixed_lines
 
 
+class FileOffsets:
+    """
+    Used to track state when applying multiple fixes to a single file/line.
+    """
+
+    def __init__(self, line_offset: int, col_offset: int, active_line: int):
+        self.line_offset = line_offset
+        self.col_offset = col_offset
+        self.active_line = active_line
```

Comment:
I think this assumes that fixes will happen in order (we will never come back and re-fix a line). This means that if one rule modifies lines 3 and 7, then another rule modifies lines 3 and 5, we'll still have a problem with line 3 (files are cached so they won't be reparsed after the autofix is applied). We should add a comment documenting this limitation.

(If not sorry for misunderstanding!)

Reply:
That makes sense.. I've extended the docstring to mention this limitation.